src/osd/PG.cc

```diff
@@ -3298,7 +3298,11 @@ void PG::reg_next_scrub()
       (info.stats.stats_invalid && g_conf->osd_scrub_invalid_stats)) {
     scrubber.scrub_reg_stamp = utime_t();
   } else {
-    scrubber.scrub_reg_stamp = info.history.last_scrub_stamp;
+    if (info.history.last_scrub_stamp + cct->_conf->osd_scrub_min_interval <
+        info.history.last_deep_scrub_stamp + cct->_conf->osd_deep_scrub_interval)
+      scrubber.scrub_reg_stamp = info.history.last_scrub_stamp;
+    else
+      scrubber.scrub_reg_stamp = info.history.last_deep_scrub_stamp;
```

Comment:
Yes, my own code only care to schedule the deep scrub in higher priority. osd should wait for min interval at least to reschedule it. But If we reschedule it immediately, pg may be rejected because of "ceph osd set nodeep-scrub". So I have not every good idea to do better.

There is another risk that other pgs may be starved of scrub. So add a new parameter such as osd_deep_scrub_check_interval(smaller than osd_scrub_min_interval) if pg is rejected by "nodeep-scrub", we should reschedule other pgs to do scrub?

I add a new issue. Maybe we could discuss this patch in http://tracker.ceph.com/issues/11387. 

Thanks for any hints.